### ISSUE TITLE
[lsp] - rename symbol keeping the same name

### ISSUE THREAD

**Issue by ericdallo** [dart-lang/sdk#42573]
It seems that `dart_analysis_server` crashes when receive a `textDocument/rename` with the same symbol name on the `newName` field.

I could not reproduce it on VSCode because vscode seems to not send the `textDocument/rename` when the symbol name is the name, but on Emacs using `lsp-dart`, we send the `textDocument/rename` and after the server returns `null` on the response(Which is ok following the LSP spec), the dart_analysis_server process crash.

Some log from the `lsp-dart` client:
```javascript
[Trace - 04:36:07 PM] Sending request 'textDocument/prepareRename - (183)'.
Params: {
  "textDocument": {
    "uri": "file:///home/greg/dev/flutter_sample/lib/screens/dashboard/components/rightscreen.dart"
  },
  "position": {
    "line": 52,
    "character": 21
  }
}

[Trace - 04:36:07 PM] Received response 'textDocument/prepareRename - (183)' in 8ms.
Result: {
  "placeholder": "_goToNextPage",
  "range": {
    "end": {
      "character": 28,
      "line": 52
    },
    "start": {
      "character": 15,
      "line": 52
    }
  }
}

[Trace - 04:36:12 PM] Sending request 'textDocument/rename - (184)'.
Params: {
  "textDocument": {
    "uri": "file:///home/greg/dev/flutter_sample/lib/screens/dashboard/components/rightscreen.dart"
  },
  "position": {
    "line": 52,
    "character": 21
  },
  "newName": "_goToNextPage"
}

[Trace - 04:36:12 PM] Received response 'textDocument/rename - (184)' in 7ms.
Result: null
```

Tested on flutter channel stable, dart 2.8.1 and flutter dev channel, `Dart 2.9.0 (build 2.9.0-19.0.dev 7e72c9ae7e)`

**Comment by ericdallo** [comment]
https://github.com/emacs-lsp/lsp-dart/issues/41

**Comment by bwilkerson** [comment]
@scheglov @DanTup

**Comment by DanTup** [comment]
I get a different response from the server than in the log above (an error instead of null):

```
Content-Length: 114
Content-Type: application/vscode-jsonrpc; charset=utf-8
{"id":7,"jsonrpc":"2.0","error":{"code":-32010,"message":"The new name must be different than the current name."}}
```

However, the server still stopped responding (even in a test). It turned out to be a silly mistake - I had reused an error code, and it happened to the the one raised then the client/server are out-of-sync which results in a server shutdown 😞

I've got a fix here: https://dart-review.googlesource.com/c/sdk/+/153341/

**Comment by ericdallo** [comment]
I see, thank you for the quick fix @DanTup, I'll keep an eye on that :)

**Comment by DanTup** [comment]
This was fixed by 34ad5a590836b9a2d679063b3f06d7a692578b31.

(@bwilkerson @ericdallo I don't have powers to close this, so someone will have to do it for me. Thanks!)

**Comment by ericdallo** [comment]
Thank you very much @DanTup :)
